adapters/repos/db/lsmkv/roaringset/serialization_test.go

```diff
@@ -0,0 +1,112 @@
+package roaringset
+
+import (
+	"bytes"
+	"math"
+	"testing"
+
+	"github.com/dgraph-io/sroar"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSerialization_HappyPath(t *testing.T) {
+	additions := sroar.NewBitmap()
+	additions.SetMany([]uint64{1, 2, 3, 4, 6})
+	deletions := sroar.NewBitmap()
+	deletions.SetMany([]uint64{5, 7})
+	key := []byte("my-key")
+
+	sn, err := NewSegmentNode(key, additions, deletions)
+	require.Nil(t, err)
+
+	buf := sn.ToBuffer()
+	assert.Equal(t, sn.Len(), uint64(len(buf)))
+
+	newSN := NewSegmentNodeFromBuffer(buf)
+	assert.Equal(t, newSN.Len(), uint64(len(buf)))
+
+	// without copying
+	newAdditions := newSN.Additions()
+	assert.True(t, newAdditions.Contains(4))
+	assert.False(t, newAdditions.Contains(5))
+	newDeletions := newSN.Deletions()
+	assert.False(t, newDeletions.Contains(4))
+	assert.True(t, newDeletions.Contains(5))
+	assert.Equal(t, []byte("my-key"), newSN.PrimaryKey())
+
+	// with copying
+	newAdditions = newSN.AdditionsWithCopy()
+	assert.True(t, newAdditions.Contains(4))
+	assert.False(t, newAdditions.Contains(5))
+	newDeletions = newSN.DeletionsWithCopy()
+	assert.False(t, newDeletions.Contains(4))
+	assert.True(t, newDeletions.Contains(5))
+}
+
+func TestSerialization_InitializingFromBufferTooLarge(t *testing.T) {
+	additions := sroar.NewBitmap()
+	additions.SetMany([]uint64{1, 2, 3, 4, 6})
+	deletions := sroar.NewBitmap()
+	deletions.SetMany([]uint64{5, 7})
+	key := []byte("my-key")
+
+	sn, err := NewSegmentNode(key, additions, deletions)
+	require.Nil(t, err)
+
+	buf := sn.ToBuffer()
+	assert.Equal(t, sn.Len(), uint64(len(buf)))
+
+	bufTooLarge := make([]byte, 3*len(buf))
+	copy(bufTooLarge, buf)
+
+	newSN := NewSegmentNodeFromBuffer(buf)
```

Comment:
shouldn't be `bufTooLarge` used to create newSN?

Reply:
Good catch, it sounds like a wrong test. Let me try to change it and see if everything still passes...